ui/src/components/Tasks/AddSample.jsx

```diff
@@ -45,12 +45,13 @@ function AddSample(props) {
     }
   }, [setFocus, show]);
 
-  function addAndMount(params) {
+  async function addAndMount(params) {
     const sampleData = getSampleData(params);
     addSamplesToList([sampleData]);
-    addSampleAndMount(sampleData);
     hide();
 
+    await addSampleAndMount(sampleData);
```

Comment:
Calling after `hide` so the modal doesn't wait for the call to succeed before disappearing (unlike `showList`, which is the important part, since we wait for the "Queued samples" tab to no longer be active in the test as an indication that the call has succeeded)